src/vsr/replica.zig

```diff
@@ -9744,7 +9810,9 @@ pub fn ReplicaType(
 
             const upgrade = vsr.UpgradeRequest{ .release = self.upgrade_release.? };
             self.send_request_to_self(.upgrade, std.mem.asBytes(&upgrade));
-            assert(self.pipeline.queue.contains_operation(.upgrade));
+
+            // This request may be rejected in on_request due to lack of space in the WAL.
+            maybe(self.pipeline.queue.contains_operation(.upgrade));
```

Comment:
Might want to add this case to `// Normally we chain send-upgrade-to-self via the commit chain.` comment